scheduler/task_finder_test.go

```diff
@@ -16,84 +16,277 @@ func init() {
+}
+
+func TestLegacyDBTaskFinder(t *testing.T) {
+	s := new(TaskFinderSuite)
+	s.taskFinder = &LegacyDBTaskFinder{}
+
+	suite.Run(t, s)
+}
+
+func (s *TaskFinderSuite) SetupTest() {
+	taskIds := []string{"t1", "t2", "t3", "t4"}
+	s.tasks = []task.Task{
+		{Id: taskIds[0], Status: evergreen.TaskUndispatched, Activated: true},
+		{Id: taskIds[1], Status: evergreen.TaskUndispatched, Activated: true},
+		{Id: taskIds[2], Status: evergreen.TaskUndispatched, Activated: true},
+		{Id: taskIds[3], Status: evergreen.TaskUndispatched, Activated: true, Priority: -1},
+	}
+
+	depTaskIds := []string{"td1", "td2"}
+	s.depTasks = []task.Task{
+		{Id: depTaskIds[0]},
+		{Id: depTaskIds[1]},
+	}
+
+	s.Nil(db.Clear(task.Collection))
+}
+
+func (s *TaskFinderSuite) insertTasks() {
+	for _, task := range s.tasks {
+		s.Nil(task.Insert())
+	}
+	for _, task := range s.depTasks {
+		s.Nil(task.Insert())
+	}
+}
+
+func (s *TaskFinderSuite) TestNoRunnableTasksReturnsEmptySlice() {
+	// XXX: collection is deliberately empty
+	runnableTasks, err := s.taskFinder.FindRunnableTasks()
+	s.Nil(err)
+	s.Empty(runnableTasks)
+}
+
+func (s *TaskFinderSuite) TestInactiveTasksNeverReturned() {
+	// insert the tasks, setting one to inactive
+	s.tasks[2].Activated = false
+	s.insertTasks()
+
+	// finding the runnable tasks should return two tasks
+	runnableTasks, err := s.taskFinder.FindRunnableTasks()
+	s.Nil(err)
+	s.Len(runnableTasks, 2)
+}
+
+func (s *TaskFinderSuite) TestTasksWithUnsatisfiedDependenciesNeverReturned() {
+	// edit the dependency tasks, setting one to have finished
+	// successfully and one to have finished unsuccessfully
+	s.depTasks[0].Status = evergreen.TaskFailed
+	s.depTasks[1].Status = evergreen.TaskSucceeded
+
+	// edit the tasks, setting one to have unmet dependencies, one to
+	// have no dependencies, and one to have successfully met
+	// dependencies
+	s.tasks[0].DependsOn = []task.Dependency{}
+	s.tasks[1].DependsOn = []task.Dependency{{s.depTasks[0].Id, evergreen.TaskSucceeded}}
+	s.tasks[2].DependsOn = []task.Dependency{{s.depTasks[1].Id, evergreen.TaskSucceeded}}
+
+	s.insertTasks()
+
+	// finding the runnable tasks should return two tasks (the one with
+	// no dependencies and the one with successfully met dependencies
+	runnableTasks, err := s.taskFinder.FindRunnableTasks()
+	s.Nil(err)
+	s.Len(runnableTasks, 2)
+}
+
+type TaskFinderComparisonSuite struct {
+	suite.Suite
+	oldRunnableTasks []task.Task
+	newRunnableTasks []task.Task
+}
+
+func TestTaskFinderComparisonSuite(t *testing.T) {
+	s := new(TaskFinderComparisonSuite)
+
+	suite.Run(t, s)
+}
+
+func (s *TaskFinderComparisonSuite) SetupTest() {
+	session, _, _ := db.GetGlobalSessionFactory().GetSession()
+	s.NotNil(session)
+	s.NoError(session.DB(taskFinderTestConf.Database.DB).DropDatabase())
+
+	for _, task := range tasks {
+		s.NoError(task.Insert())
+	}
+
+	oldTaskFinder := &LegacyDBTaskFinder{}
+	newTaskFinder := &DBTaskFinder{}
+	var (
+		err_old error = nil
+		err_new error = nil
```

Comment:
you don't need to specify the zero value for a type (for error, it's nil) when you declare variables.